pkg/k8s/service.go

```diff
@@ -145,8 +145,16 @@ func ParseService(svc *slim_corev1.Service, nodeAddressing datapath.NodeAddressi
 				proto := loadbalancer.L4Type(port.Protocol)
 				port := uint16(port.NodePort)
 				id := loadbalancer.ID(0) // will be allocated by k8s_watcher
-
+				if !option.Config.EnableIPv4 &&
```

Comment:
In an ideal world I'd prefer to make `pkg/k8s/service.go` free from `option.Config.EnableIPv{4,6}`, and let it parse service objects regardless of daemon settings. Then, `pkg/service` would take care (by looking into the settings) of whether to provision a given service.

Reply:
So, what happens when the service object is partially valid (i.e. it has both ipv{4,6}, but only 4 is enabled)?

Reply:
Then I'd log a warning and ignore the v6 svc.

Reply:
@brb I don't understand your comment, this file already has `option.Config.EnableIPv4` (see line 156 for example)

Reply:
@aanm My point is that we should move all `option.Config.*` from `ParseService()` to `pkg/service` instead of adding some logic to the function which supposed just to convert a k8s service obj to an internal service representation.

Reply:
There are other `option.Config.*` instances in this function though. The work of pulling those out of `ParseService` is enough work that I think it can be fairly deemed out of scope.

Reply:
> There are other `option.Config.*` instances in this function though. The work of pulling those out of `ParseService` is enough work that I think it can be fairly deemed out of scope.

Yes, I totally agree with this.